Bento/Decorators/CustomInputComponent.swift

```diff
@@ -9,37 +10,66 @@ extension Renderable {
         return CustomInputComponent(
             source: self,
             customInput: input,
-            contentStatus: contentStatus,
-            highlightColor: highlightColor
+            focusEligibility: .eligible(contentStatus),
+            highlightColor: highlightColor,
+            focusesOnFirstDisplay: false
+        ).asAnyRenderable()
+    }
+
+    /// Sets up `customInput` to prepare for presention when user taps `self`,
+    /// and also presents it immediately when `state` is `.some`.
+    ///
+    /// - important: This method is useful when `customInput` needs to be displayed asynchronously after state change.
+    /// - note: Due to asynchronous presentation, `focusEligibility` is not supported.
+    public func customInputImmediately<State>(
```

Comment:
Also, I find the name confusing because this won't show immediately since it first triggers some action and only then displays the custom input. I think using the same name would be better because the parameters are clear enough to distinguish between both behaviours, either you have your input available or you are conditionally waiting for some state to get it.

Reply:
https://github.com/Babylonpartners/Bento/pull/182#discussion_r292862688

> `func customInput(immediatelyWhen:...)`

How about this?

Reply:
Renamed to `func customInput(immediatelyWhen:...)` so that it becomes more clear that `customInput` is always placed as a virtual node.